src/misc/deployHelpers.ts

```diff
@@ -0,0 +1,115 @@
+import Command from '@oclif/command';
+import * as FormData from 'form-data';
+import * as fs from 'fs';
+import fetch from 'node-fetch';
+import { Response } from 'node-fetch';
+import { AppCompiler, AppPackager, FolderDetails } from '.';
+import { IServerInfo1, IServerInfo2 } from './interfaces';
+
+export const checkReport = (command: Command, fd: FolderDetails, flags: { [key: string]: any }): void => {
+        const compiler = new AppCompiler(command, fd);
+        const report = compiler.logDiagnostics();
+
+        if (!report.isValid && !flags.force) {
+            throw new Error('TypeScript compiler error(s) occurred');
+        }
+        return;
+};
+
+export const getServerInfo = async (fd: FolderDetails): Promise<IServerInfo1 | IServerInfo2> => {
+    return new Promise((resolve, reject) => {
+        fs.readFile(fd.mergeWithFolder('serverInfo.json'), 'utf8', (error, data) => {
+            if (error) {
+                reject(error);
+            }
+            resolve(JSON.parse(data));
```

Comment:
If the `data` is an invalid JSON string, an `Unexpected token <` error may occur here. Please add try...catch block around `JSON.parse` to make your code more robust.